src/server.cpp

```diff
@@ -2043,7 +2043,16 @@ void Server::SendActiveObjectRemoveAdd(RemoteClient *client, PlayerSAO *playersa
 
 	std::queue<u16> removed_objects, added_objects;
 	m_env->getRemovedActiveObjects(playersao, my_radius, player_radius,
-		client->m_known_objects, removed_objects);
+		client->m_known_objects,
+		[&](bool gone, u16 id) {
```

Comment:
IMO this callback use is unnecessary
I would rather turn `removed_objects` into a `std::queue<std::pair<bool, u16>>` or separate the queue into removed and deleted objects

Reply:
What `getRemovedActiveObjects` does is "filters" the active objects. IMO, ideally, `getRemovedActiveObjects` would return an "iterator" (in the "streaming" sense of the word) of pairs of `bool, u16`. But C++ doesn't have such "streaming" iterators, so I like to use callbacks (instead of filling collections), esp. if I must process all items; there is no need to store these bools (though there is also no significant cost to doing it).

In C++ 20, we will be able to use [filter](https://en.cppreference.com/w/cpp/ranges/filter_view) for this :upside_down_face: 

By the way: Do you know why we use a queue - is order important here (otherwise I'd just use a vector-based stack)? Is there a way to "batch" the sound stop packets?

Anyways, turned it into a queue of pairs: https://github.com/minetest/minetest/pull/14436/commits/23c65b6c9339aceea72a404a9efee3ad0fe3bcab

Reply:
I don't see a reason for that a `std::queue` indeed.
Don't bother changing that however, I decided that I'll open a PR to clean up this code myself.